encord_agents/tasks/runner/sequential_runner.py

```diff
@@ -208,66 +207,94 @@ def decorator(func: DecoratedCallable) -> DecoratedCallable:
-                                        assert (
-                                            context.label_row is not None
-                                        ), f"Label row is not set for task {task} setting the priority requires either setting the `will_set_priority` to True on the stage decorator or depending on the label row."
-                                        context.label_row.set_priority(
-                                            agent_response.label_row_priority, bundle=label_bundle
-                                        )
-                                else:
-                                    pathway_to_follow = agent_response
-                                if pathway_to_follow is None:
-                                    pass
-                                elif next_stage_uuid := try_coerce_UUID(pathway_to_follow):
-                                    if next_stage_uuid not in [pathway.uuid for pathway in stage.pathways]:
-                                        raise PrintableError(
-                                            f"No pathway with UUID: {next_stage_uuid} found. Accepted pathway UUIDs are: {[pathway.uuid for pathway in stage.pathways]}"
-                                        )
-                                    task.proceed(pathway_uuid=str(next_stage_uuid), bundle=task_bundle)
-                                else:
-                                    if pathway_to_follow not in [str(pathway.name) for pathway in stage.pathways]:
-                                        raise PrintableError(
-                                            f"No pathway with name: {pathway_to_follow} found. Accepted pathway names are: {[pathway.name for pathway in stage.pathways]}"
-                                        )
-                                    task.proceed(pathway_name=str(pathway_to_follow), bundle=task_bundle)
-                                if pbar_update is not None:
-                                    pbar_update(1.0)
-                                break
-
-                            except KeyboardInterrupt:
-                                raise
-                            except PrintableError:
-                                raise
-                            except Exception:
-                                print(f"[attempt {attempt+1}/{num_retries+1}] Agent failed with error: ")
-                                traceback.print_exc()
+                        with ThreadPoolExecutor() as executor:
+                            dependency_list = list(
+                                executor.map(
+                                    lambda context: solve_dependencies(
+                                        context=context, dependant=runner_agent.dependant, stack=stack
+                                    ),
+                                    batch,
+                                )
+                            )
```

Comment:
I don't see why we'd want to wait for all tasks to be fetched before starting the compute? Can't we just call the agent when iterating the output of the `executor.map` call rather than wrapping it in `list`?